platforms/core-runtime/base-services/src/main/java/org/gradle/util/internal/VersionNumber.java

```diff
@@ -263,28 +263,24 @@ public String remainder() {
     }
 
     private static class DefaultScheme extends AbstractScheme {
-        private static final String VERSION_TEMPLATE = "%d.%d.%d%s";
-
         public DefaultScheme() {
             super(3);
         }
 
         @Override
         public String format(VersionNumber versionNumber) {
-            return String.format(VERSION_TEMPLATE, versionNumber.major, versionNumber.minor, versionNumber.micro, versionNumber.qualifier == null ? "" : "-" + versionNumber.qualifier);
+            return String.format("%d.%d.%d%s", versionNumber.major, versionNumber.minor, versionNumber.micro, versionNumber.qualifier == null ? "" : "-" + versionNumber.qualifier);
         }
     }
 
     private static class SchemeWithPatchVersion extends AbstractScheme {
-        private static final String VERSION_TEMPLATE = "%d.%d.%d.%d%s";
-
         private SchemeWithPatchVersion() {
             super(4);
         }
 
         @Override
         public String format(VersionNumber versionNumber) {
-            return String.format(VERSION_TEMPLATE, versionNumber.major, versionNumber.minor, versionNumber.micro, versionNumber.patch, versionNumber.qualifier == null ? "" : "-" + versionNumber.qualifier);
+            return String.format("%d.%d.%d.%d%s", versionNumber.major, versionNumber.minor, versionNumber.micro, versionNumber.patch, versionNumber.qualifier == null ? "" : "-" + versionNumber.qualifier);
```

Comment:
Why duplicate the string here?

Reply:
Not duplicate, inline. It was only used here.